### PR TITLE
fix: asciidoctor warnings and errors

### PR DESCRIPTION
**Pull request by cynful**
I'm still not sure what this error is 🤷‍♀️ 
```bash
# ruby convert.rb
asciidoctor: ERROR: sitemap.adoc: line 9: Failed to generate image: PlantUML image generation failed: [From <input> (line 2) ]
                        
@startuml               
                        
^^^^^                   
 Empty description      

Sitemap generated successfully.
```

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -6,7 +6,7 @@ ifdef::env-github[]
 :warning-caption: :warning:
 endif::[]
 
-= code-coffee-compendium image:https://img.shields.io/badge/License-MIT-yellow.svg[MIT License, link=https://opensource.org/licenses/MIT] image:https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square[PRs Welcome, link=http://makeapullrequest.com] image:https://img.shields.io/badge/View%20on-GitHub-orange[View on GitHub, link=https://github.com/LearnTeachCode/code-coffee-compendium/]
+== code-coffee-compendium image:https://img.shields.io/badge/License-MIT-yellow.svg[MIT License, link=https://opensource.org/licenses/MIT] image:https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square[PRs Welcome, link=http://makeapullrequest.com] image:https://img.shields.io/badge/View%20on-GitHub-orange[View on GitHub, link=https://github.com/LearnTeachCode/code-coffee-compendium/]
```
Comment (capsulecorplab) [inline README.adoc:9]: i'm not sure we want to bump this an extra level

Comment (cynful) [reply README.adoc:9]:
```
# ruby convert.rb
asciidoctor: WARNING: header.adoc: line 6: section title out of sequence: expected level 1, got level 2
asciidoctor: WARNING: header.adoc: line 6: section title out of sequence: expected level 1, got level 2
asciidoctor: WARNING: header.adoc: line 6: section title out of sequence: expected level 1, got level 2
asciidoctor: ERROR: README.adoc: line 9: level 0 sections can only be used when doctype is book
```
we get this ERROR, not warning here, `level 0 sections can only be used when doctype is book`

Comment (cynful) [reply README.adoc:9]: ^ in this case I think its a style preference for asciidoctor, sim. to python black, which is strict

Comment (cynful) [reply README.adoc:9]: visually it seems fine at level 1
Screen Shot 2020-05-21 at 10 57 39 AM

**Comment by capsulecorplab** [comment]
> I'm still not sure what this error is 
> 
> ```shell
> # ruby convert.rb
> asciidoctor: ERROR: sitemap.adoc: line 9: Failed to generate image: PlantUML image generation failed: [From <input> (line 2) ]
>                         
> @startuml               
>                         
> ^^^^^                   
>  Empty description      
> 
> Sitemap generated successfully.
> ```

Two issues possibly pertaining to the above:

- plantuml may need an "entry" node that points to (in our case) index see https://github.com/plantuml/plantuml/issues/212
- docker's asciidoctor isn't loading the asciidoctor-diagram plugin

File: README.adoc (hunk 1 above)
Comment (cynful) [reply README.adoc:9]: > Hypotheses possibly pertaining to the above:
> 
> * plantuml may need an "entry" node that points to (in our case) index see [plantuml/plantuml#212](https://github.com/plantuml/plantuml/issues/212)
> * docker's asciidoctor may not be loading the asciidoctor-diagram plugin https://hub.docker.com/r/asciidoctor/docker-asciidoctor/
> * `convert.rb` may not be calling the asciidoctor-diagram extension https://asciidoctor.org/docs/user-manual/#ruby-api-options

```bash
$ docker run -it -v /localpath/code-coffee-compendium:/documents/ asciidoctor/docker-asciidoctor:latest
bash-5.0# asciidoctor -r asciidoctor-diagram sitemap.adoc
```
confirms this works and does generate the diagram
rules out number 2

more likely number 3, line 44 does not call ascii-doctor-diagram
also script errors (keeps going) 
@TristanDamron, can we either remove sitemap.adoc from the set that gets converted?
or can you fix line 44 to also convert with -r asciidoctor-diagram sitemap.adoc

```
# ruby convert.rb 
asciidoctor: ERROR: sitemap.adoc: line 9: Failed to generate image: PlantUML image generation failed: [From <input> (line 2) ]
                        
@startuml               
                        
^^^^^                   
 Empty description      

Sitemap generated successfully.
```

**Pull request merged**